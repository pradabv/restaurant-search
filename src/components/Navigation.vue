<template>
   <div>
    <h3>{{business.length}} {{business.length >1  ? 'Restaurants' : 'Restaurant'}}</h3>
      <div class="restaurant-list">
        <ul>
            <li @click="selectItem(item)" v-for="(item, index) in business" :key="index" :class=" selectedItem && selectedItem.id === item.id ? 'active' : ''">
                <a>
                    <h2>{{item.name}}</h2>
                    <h4 class="location">
                        <img src="../assets/icons/marker.svg" alt="restaurant location" title="restaurant location">
                        <span class="address" v-if="item.location.address1">{{item.location.address1}},</span>
                        <span v-if="item.location.city">{{item.location.city}}</span>
                    </h4>
                </a>
            </li>
        </ul>
      </div>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import store from '@/store';
import { AppTypes } from '@/types/apptypes';

@Component
export default class Navigation extends Vue {
  get business(): AppTypes.Business[] {
    return store.state.business;
  }

  get selectedItem() {
    return store.state.selectedItem;
  }

  selectItem(item: AppTypes.Business) {
    store.commit('setSelectedItem', item);
  }
  
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss">
.main-left h3{
    background-color: #d3d3d3;
    font-size: 16px;
    padding: 15px 30px;
}
.restaurant-list{
    height: 74vh;
    overflow-y: scroll;
    overflow-x: hidden;
}

.restaurant-list h2{
    font-size: 16px;
    margin-bottom: 10px;
}
h4.location{
    font-size: 15px;
    font-weight: normal;
    flex-direction: row;
    box-sizing: border-box;
    display: flex;
    place-content: flex-start;
    align-items: flex-start;
}
h4.location img{
    width: 20px; 
    height: auto;
    margin-right: 5px;
}
.restaurant-list ul{
    margin: 0px; 
    padding: 0px;
}
.restaurant-list ul li{
    list-style: none;
    padding: 24px 30px;
    border-bottom:#d3d3d3 1px solid;
}
.restaurant-list ul li.active{
    background-color: #61dafb;
}
.restaurant-list ul li:hover{
    background-color: #e8faff;
}
.restaurant-list ul li.active:hover{
    background-color: #61dafb;
}
.restaurant-list ul li a{
    color: #282c34;
}
.restaurant-list ul li a:hover{
    color: #282c34;
}
.location .address{
  margin-right: 5px;
}
</style>
