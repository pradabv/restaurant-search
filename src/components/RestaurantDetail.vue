<template>
    <div>
        <div v-if="selectedItem">
            <div class="restaurant-details">
                <div class="retaurant-header">
                    <h2><a :href="selectedItem.url" :target="'_blank'">{{selectedItem.name}}</a></h2>
                    <p><b>Rating:</b> {{selectedItem.rating}}</p>
                    <h4 class="location">
                        <img src="../assets/icons/marker.svg" alt="restaurant location" title="restaurant location">
                        <span>{{selectedItem.location.formatted_address}}</span>
                    </h4>
                    <h4 class="location" v-if="selectedItem.display_phone">
                        <img src="../assets/icons/phone.svg" alt="restaurant number" title="restaurant number">
                        <span>{{selectedItem.display_phone}}</span>
                    </h4>
                </div>
                <div class="retaurant-img" v-for="(img, index) of selectedItem.photos" :key="index"> 
                    <img :src="img" :alt="selectedItem.name" :title="selectedItem.name">
                </div>
            </div>
            <div class="restaurant-reviews">
                <h5>Reviews</h5>
                <ul v-if="selectedItem.reviews.length">
                    <li v-for="(review, index) in selectedItem.reviews" :key="index"> 
                        <p><b>Rating:</b> {{review.rating}}</p>
                        <p>{{review.text}}</p>
                    </li>
                </ul>
                <p v-else>No reviews found</p>
            </div>
        </div>
        <div v-else class="welcome-screen">
            <h3>Welcome to Restaurant Finder</h3>
            <p>Please Select a Restaurant to view details</p>
        </div>
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import store from '@/store';

@Component
export default class RestaurantDetail extends Vue {
  mounted() {
    store.dispatch('getJsonData');
  }

  get selectedItem() {
    return store.state.selectedItem;
  }

}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss">

.restaurant-details{
    flex-direction: row;
    box-sizing: border-box;
    display: flex;
    place-content: flex-start;
    align-items: flex-start;
    padding: 40px;
}
.welcome-screen{
   padding: 15px 250px;
}
.retaurant-header{
    width: 70%;
}
.retaurant-header h2{
    text-decoration: underline;
    margin-bottom: 20px;
}
.retaurant-header a {
    color:  #282c34;
}
.retaurant-header h4.location{
    font-size: 16px;
    margin: 20px 0px;
    font-weight: normal;
}

.retaurant-img{
    width: 30%;
}
.retaurant-img img{
    width: 100%;
    height: auto;
}

.restaurant-reviews{
    padding: 0px 40px;
}
.restaurant-reviews h5{
    font-size: 20px;
}
.restaurant-reviews ul{
    height: 30vh;
    padding-right: 30px;
    overflow-y: auto;
    overflow-x: hidden;
}
</style>


