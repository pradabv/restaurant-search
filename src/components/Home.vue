<template>
  <div>
    <Header  appHeaderText="Restaurant Finder" />
    <section class="main-section">
      <div class="main-left">
        <Navigation />
      </div>
      <div class="main-right">
        <RestaurantDetail />
      </div>
    </section>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import Navigation from './Navigation.vue';
import  Header from './Header.vue';
import  RestaurantDetail from './RestaurantDetail.vue';
import store from '@/store';

@Component({
    components: {
      Header,
      Navigation,
      RestaurantDetail
    }
})
export default class Home extends Vue {
  mounted() {
    store.dispatch('getJsonData')
  }

  get selectedItem() {
    return store.state.selectedItem;
  }

}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss">
.main-section{
    flex-direction: row;
    box-sizing: border-box;
    display: flex;
    place-content: flex-start;
    align-items: flex-start;
}


.main-left{
    width: 30%;
    border-right:#d3d3d3 1px solid;
}
.main-right{
    width: 70%;
}

</style>


