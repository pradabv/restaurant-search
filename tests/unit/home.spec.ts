
import { shallowMount } from '@vue/test-utils';
import Home from '@/components/Home.vue';
import Navigation from '@/components/Navigation.vue';
import RestaurantDetail from '@/components/RestaurantDetail.vue';

describe('Home.vue', () => {
  it('Home check its children component', () => {
    const wrapper = shallowMount(Home);
    expect(wrapper.findComponent(Navigation).html()).toBeTruthy();
    expect(wrapper.findComponent(RestaurantDetail).html()).toBeTruthy();
  });

  it('Home MatchSnapshot', () => {
    const wrapper = shallowMount(Home);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
  