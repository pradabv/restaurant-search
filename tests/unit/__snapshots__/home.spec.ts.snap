// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue Home MatchSnapshot 1`] = `
<div>
  <header-stub appheadertext="Restaurant Finder"></header-stub>
  <section class="main-section">
    <div class="main-left">
      <navigation-stub></navigation-stub>
    </div>
    <div class="main-right">
      <restaurantdetail-stub></restaurantdetail-stub>
    </div>
  </section>
</div>
`;
